### DISCUSSION TITLE
Check if the argument provided is a function & not a Mongoose model

### DISCUSSION THREAD

**Discussion by samislam** [Automattic/mongoose#12056] [Q&A] [answered]
I am writing a middleware. The middleware accepts two arguments, either a mongoose Model or a normal JavaScript function. 

```js
function someMiddleware(arg1){
   // here, how to differentiate?
   if(typeof arg1 === 'function') console.log("it's a function")
}
```

Mongoose models are functions, but I want the consumer of my **someMiddleware** function to be able to pass a model or a normal function. 

But how can I differentiate between both in my code?

asked also on StackOverflow, but did not get a fulfilling answer:

https://stackoverflow.com/questions/72874584/check-if-the-argument-provided-is-a-function-not-a-mongoose-model

**Comment by vkarpov15** [comment, marked as answer]
`Object.getPrototypeOf(arg1) !== mongoose.Model`
